### PR TITLE
Almayer AntiTheft measures.

### PR DESCRIPTION
**Pull request by realforest2001**
STOP STEALING SHIT
# About the pull request
Adds a subtype of reinforced hull for the Almayer that changes between indestructible hull and normal reinforced wall after hijack.
Uses this wall type around uniform vendors, the separation wall in the firing range and around engineering storage.

# Explain why it's good for the game
Having the engineering storage looted at round start is just silly avoidance of the "don't deconstruct the whole ship without a reason" rule.

# Testing Photographs and Procedure

I have verified that all works as intended, the walls cannot be harmed prior to hijack, and shutters function as advertised.

# Changelog
:cl:
add: Added a new almayer hull type (heavy reinforced) which is indestructible by normal means until after hijack collision.
add: Added a new subtype of shutter that automatically opens or closes depending on security level.
maptweak: Maps both of the above around the engineering storeroom. Also adds the walls between the firing ranges and around uniform vendors.
maptweak: Manual control button for shutters over engineering storage in the CEs office.
code: Changes hijack structural changes (walls/windows/windoors/ladders) to use signals.
/:cl:

### CODE REVIEW

File: code/modules/shuttle/dropship_hijack.dm [hunk 1]
```diff
@@ -54,6 +54,9 @@
 		var/obj/structure/window/H = i
 		H.deconstruct(FALSE)
 
+	for(var/turf/closed/wall/almayer/reinforced/temphull/hull in GLOB.hijack_bustable_hull)
```
Comment (fira) [inline code/modules/shuttle/dropship_hijack.dm:57]: as anything

would have been better to use a SSdcs signal here so other objects can use it too

File: code/modules/shuttles/marine_ferry.dm [hunk 2]
```diff
@@ -458,6 +458,9 @@
 		var/obj/structure/window/H = i
 		H.deconstruct(FALSE)
 
+	for(var/turf/closed/wall/almayer/reinforced/temphull/hull in GLOB.hijack_bustable_hull)
```
Comment (fira) [inline code/modules/shuttles/marine_ferry.dm:461]: as anything

**Review by fira — CHANGES_REQUESTED** [review]

**Review by fira — APPROVED** [review]

File: code/game/machinery/doors/shutters.dm [hunk 3]
```diff
@@ -155,3 +155,24 @@
 	if(HAS_TRAIT(attacking_item, TRAIT_TOOL_CROWBAR))
 		return
 	..()
+
+/obj/structure/machinery/door/poddoor/shutters/almayer/uniform_vendors/antitheft
```
[the remark this answers is not in the document]
Comment (realforest2001) [reply code/game/machinery/doors/shutters.dm:159]: Yah, pre-existing overrides to damage stuff on /uniform_vendors

**Comment by realforest2001** [comment]
> i think making indestructible walls on engineering may not be a good idea. adding too much mechanical protection makes MPs a lil useless dont you think ? . the best would be first to make sure only MTs and CE have access to the Storage room and maybe using reinforced Doors to prevent easy hacking

It already has reinforced doors and MT/CE access only. Insufficient to deter people rushing it at round start, so here we are.

**Comment by zzzmike** [comment]
> adding too much mechanical protection makes MPs a lil useless dont you think ?

As someone with a lot of MP hours, I'm not concerned at all with this change. There is so, so much more to do.

**Review by Nanu308 — APPROVED** [review]
Mapping seems fine

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Pull request merged**